rotkehlchen/exchanges/htx.py

```diff
@@ -188,13 +204,135 @@ def query_balances(self, **kwargs: Any) -> ExchangeQueryBalances:
 
         return returned_balances, ''
 
+    def _paginated_query(
+            self,
+            endpoint: Literal[
+                '/v1/query/deposit-withdraw',
+                '/v1/order/matchresults',
+            ],
+            options: dict[str, Any],
+    ) -> list[dict[str, Any]]:
+        """
+        Query endpoints that can return paginated data. At the moment this is used for asset
+        movements. For asset movements we use cursor pagination since for any time range
+        the api will return a valid response and we don't know when to stop querying.
+        The cursor pagination uses the id of the last object in the last
+        query to return the results of the next page.
+        """
+        result = []
+        # copy since we are modifying the dict and could affect other queries using
+        # the same options
+        query_options = options.copy()
+        while True:
+            output = self._query(absolute_path=endpoint, options=query_options)
+            if len(output) == 0:
+                break
+
+            query_options['from'] = output[-1]['id']
+
+            result.extend(output)
+            if len(output) < PAGINATION_LIMIT:
+                break
+
+        return result
+
+    def _query_deposits_withdrawals(
+            self,
+            start_ts: Timestamp,
+            end_ts: Timestamp,
+            query_for: AssetMovementCategory,
+    ) -> list[AssetMovement]:
+        """
+        Process deposits/withdrawals from HTX. If any asset is unknown or we fail to
+        deserialize an entry the error is logged and we skip the entry.
+
+        This logic doesn't filter by timestamp and instead processes all the entries
+        returned by the api discarding those that aren't in start_ts <= timestamp <= end_ts.
+        The reason is that the api is poorly designed and you can't know until when to query,
+        for any time range the api will return values and is not possible to know when the user
+        started to use the api.
+        """
+        log.debug(f'querying HTX online with {start_ts=}-{end_ts=}')
+        raw_data = self._paginated_query(
+            endpoint='/v1/query/deposit-withdraw',
+            options={
+                'type': 'deposit' if query_for == AssetMovementCategory.DEPOSIT else 'withdraw',
+                'size': PAGINATION_LIMIT,
+            },
```

Comment:
is this one of those api that doesn't allow to set a start date?

Reply:
Yes, https://www.htx.com/en-us/opend/newApiPages/?id=7ec4f050-7773-11ed-9966-0242ac110003